alf/environments/carla_sensors.py

```diff
@@ -2152,3 +2157,402 @@ def get_current_observation(self, current_frame):
             np.ndarray: 1D vector contains the distance to the frontal obstacle.
         """
         return np.array([self._distance]).astype(np.float32)
+
+
+# ==============================================================================
+# -- Dynamic Object Sensor -----------------------------------------------------
+# ==============================================================================
+
+
+@alf.configurable
+class DynamicObjectSensor(SensorBase):
+    """DynamicObjectSensor.
+        A sensor that perceives the dynamic objects around the ego agent.
+    """
+
+    def __init__(self,
+                 parent_actor,
+                 alf_world,
+                 history_idx=[-16, -11, -6, -1],
+                 max_history_len=20,
+                 object_filter='vehicle.*',
+                 max_object_number=3,
+                 distance_threshold=100,
+                 with_ego_hist=True,
+                 fill_value=None):
+        """
+        Args:
+            parent_actor (carla.Actor): the parent actor of this sensor
+            alf_world (World): the world object keeping all relevant data and
+                some utility functions (e.g., `_get_traffic_light_waypoints`)
+            navigation_sensor (str): the navigation sensor associated with the
+                `parent_actor`
+            history_idx (list[int]): a list of numbers representing the indices
```

Comment:
Really like this interface design of using `history_idx` list. Question: can `max_history_len` be inferred from this?

Reply:
Good point. Removed ``max_history_len`` since it can be inferred.